keys/src/address.rs

```diff
@@ -159,16 +159,33 @@ impl From<Blake2bHash> for Address {
     }
 }
 
-impl<'a> From<&'a PublicKey> for Address {
-    fn from(public_key: &'a PublicKey) -> Self {
+impl<'a> From<&'a EdDSAPublicKey> for Address {
+    fn from(public_key: &'a EdDSAPublicKey) -> Self {
+        let hash = Blake2bHasher::default().digest(public_key.as_bytes());
+        Address::from(hash)
+    }
+}
+
+impl<'a> From<&'a WebauthnPublicKey> for Address {
+    fn from(public_key: &'a WebauthnPublicKey) -> Self {
         let hash = Blake2bHasher::default().digest(public_key.as_bytes());
         Address::from(hash)
     }
 }
```

Comment:
Is this a problem? We now effectively assign an address to double the amount of possible keys.
We have 2^160 addresses. Previously we had 2^256 keys, thus 2^96 keys per address.
Now it would be 2^257 keys, thus 2^97 keys per address.

Reply:
Due to the multi sig scheme / merkle path, every key is essentially valid for every address, you "just" have to find the bytes to combine with your public key to make the the cosigners merkle root match the address.

What matters is not how many keys there are per address, but the probability of a collision, which is not affected by this.